tests/integration/servicemesh/federation/discovery/discovery_test.go

```diff
@@ -58,7 +58,7 @@ var (
 	// Timeout is 6 minutes long, because we have hardcoded resync period 5 minutes long in the federation discovery controller.
 	defaultRetry = echo.Retry{
 		Options: []retry.Option{
-			retry.Timeout(6 * time.Minute),
+			retry.Timeout(30 * time.Second),
```

Comment:
should the timeout be based on resync period? maybe min(resync, 30s)?

Reply:
I noticed that the first test is sometimes slow and takes ~12 seconds, so there must be also other factors to tweak. Our patches rarely affect federation, so these tests should not fail, and therefore I think 30 seconds long timeout is a good trade-off.